hq/app/logic/ReportDisplay.scala

```diff
@@ -91,6 +92,17 @@ object ReportDisplay {
     ReportSummary(warnings, errors, other)
   }
 
+  def exposedKeysSummary(allReports: Map[AwsAccount, Either[FailedAttempt, List[ExposedIAMKeyDetail]]]): Map[AwsAccount, Boolean] = {
+    def checkForExposedKeys(temp: Either[FailedAttempt, List[ExposedIAMKeyDetail]]): Boolean = temp match {
+      case Right(keys) if keys.nonEmpty => true
+      case _ => false
+    }
+
+    for {
+      report <- allReports
```

Comment:
You should be able to extract the tuple parts with good names here:

```scala
for {
  (account, report) <- allReports
} yield (account, checkForExposedKeys(report))
```

Reply:
Much nicer code, thank you 🎉